source/lac/cuda_vector.cu

```diff
@@ -916,6 +916,14 @@ namespace LinearAlgebra
 
 
     template <typename Number>
+    bool Vector<Number>::all_zero() const
+    {
+      return (linfty_norm() == 0) ? true : false;
```

Comment:
I will need to launch as many threads as entries anyway, so I don't see how I can gain anything from exiting early. Also I thought that sending the data back to the host will take a large amount of time, so there is no point on doing something complicated for this.